### PR TITLE
DUPLO-11667: TF: duplocloud_admin_system_setting.DBInstanceParameter gives Provider produced inconsistent result after apply error

### PR DESCRIPTION
**Pull request by pravin-duplocloud**
## Overview

DUPLO-11667: TF: duplocloud_admin_system_setting.DBInstanceParameter gives Provider produced inconsistent result after apply error

## Summary of changes

This PR does the following:

- Escape '/' in terrform id

## Testing performed

- [ ] Using unit tests
- [X] Manually, on my local system
- [X] Manually, on a remote test system

## Describe any breaking changes

- None

___

## **Type**
bug_fix, enhancement

___

## **Description**
- Fixed an issue with handling slashes in `keyType` for resource ID creation and parsing in `duplocloud_admin_system_setting`.
- Added utility functions to encode and decode slashes in identifiers, enhancing the handling of special characters.
- Updated build and import scripts to include documentation generation and guide on handling special characters.
- Updated CHANGELOG and documentation to reflect these changes.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_admin_system_setting.go</strong><dd><code>Handle Slashes in Admin System Setting IDs</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplocloud/resource_duplo_admin_system_setting.go

<li>Escaped slashes in <code>keyType</code> during ID creation for admin system <br>settings.<br> <li> Decoded slashes in <code>keyType</code> when parsing admin system setting IDs.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/510/files#diff-be1501b5adb17b6e67f12db16c4aa145b25127bf58200c2aa7cf78a9bffe5243">+3/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>utils.go</strong><dd><code>Enhance Identifier Handling with Slash Encoding/Decoding</code>&nbsp; </dd></summary>
<hr>
      
duplosdk/utils.go

<li>Added functions to encode and decode slashes in identifiers.<br> <li> Added logging for replacements in identifiers.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/510/files#diff-c5d81c3dd5cc18373553e9b27fb08e4366b1024b38797fa266fb27c50aef6fb1">+18/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>build.sh</strong><dd><code>Update Build Script to Include Documentation Generation</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
build.sh

- Added documentation generation command to the build script.

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/510/files#diff-4d2a8eefdf2a9783512a35da4dc7676a66404b6f3826a8af9aad038722da6823">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>import.sh</strong><dd><code>Update Import Script for Admin System Setting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/resources/duplocloud_admin_system_setting/import.sh

<li>Updated the import script to instruct on replacing '/' with '_SLASH_' <br>in <code>KEY_TYPE</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/510/files#diff-00ed156dc60525af9c21bf34c224170e5d206606641916d005ac85ec61e9bac1">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update Changelog for Slash Handling Fixes and Enhancements</code></dd></summary>
<hr>
      
CHANGELOG.md

<li>Documented fixes and enhancements regarding slash handling in resource <br>IDs.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/510/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>admin_system_setting.md</strong><dd><code>Document Slash Replacement in Admin System Setting Import</code></dd></summary>
<hr>
      
docs/resources/admin_system_setting.md

<li>Updated documentation to guide on replacing '/' with '_SLASH_' in <br><code>KEY_TYPE</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/510/files#diff-2f04aad908bac31f7d316630117291570133799837aadabc32ca8afb2b227f72">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-11667 TF: duplocloud_admin_system_setting.DBInstanceParameter gives Provider produced inconsistent result after apply error](https://app.clickup.com/t/8655600/DUPLO-11667)

File: examples/resources/duplocloud_admin_system_setting/import.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Example: Importing an existing duplo admin setting
-#  - *KEY_TYPE* is the type of setting key
+#  - *KEY_TYPE* is the type of setting key. Replace any occurrences of '/' with 'SLASH' if they exist within the KEY_TYPE.
```
Comment (tahir-duplo) [inline examples/resources/duplocloud_admin_system_setting/import.sh:2]: Here it should be _SLASH_ right  instead of SLASH?

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Review by Duplo-nikv — APPROVED** [review]

File: examples/resources/duplocloud_admin_system_setting/import.sh (hunk 1 above)
Comment (pravin-duplocloud) [reply examples/resources/duplocloud_admin_system_setting/import.sh]: done

**Pull request merged**